app/jobs/mailers/monthly_stats.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Mailers
+  class MonthlyStats < ApplicationJob
+    queue_as :mailers
+
+    def perform
+      User.where(provider: 'azure_oauth', opt_out_stats_email: false, active: true).each do |user|
+        ActorMailer.with(actor: user.actor).monthly_stats.deliver_later
```

Comment:
When you call `user.actor` here, inside the loop, it is hitting the database for the associated actor once per iteration of the loop. This is what's commonly called a "n+1 query bug" because it hits the database once to load out all the Users, then _n_ times for every iteration to load the Actor. The way to avoid this is in Rails-speak is to change the line above from `User.where(...)` to `User.includes(:actor).where(...)`. This tells rails to automatically inner-join the actors table and load everything out at once, so you only have a single database hit.